library/Cube/DbCube.php

```diff
@@ -165,7 +164,12 @@ public function finalizeInnerQuery()
             }
         }
 
-        $query->columns($columns + $this->factColumns);
+        $c = [];
+        foreach ($columns + $this->factColumns as $k => $v) {
+            $c[$this->connection->getDbAdapter()->quoteIdentifier($k)] = $v;
```

Comment:
Either use `$this->connection->getDbAdapter()` or just `$this->db()` but not both.